File: examples/example-kubernetes-client-openapi-operator-python/src/main/python/micronaut/operator/SecretOperator.py
# tag::class[]
import logging

from micronaut.context.annotation import Context
from micronaut.kubernetes.client.openapi.informer import SharedIndexInformerFactory
from micronaut.kubernetes.client.openapi.model import V1Secret
from micronaut.kubernetes.client.openapi.operator import OperatorResourceLister
from micronaut.kubernetes.client.openapi.operator.controller import ControllerFactory
from micronaut.kubernetes.client.openapi.operator.controller.reconciler import Request, ResourceReconciler, Result
from micronaut.kubernetes.client.openapi.resolver import NamespaceResolver
# end::class[]
from micronaut.context.annotation import Requires
from micronaut.context.env import Environment
# tag::class[]

LOG = logging.getLogger(__name__)

# end::class[]

@Requires(env=Environment.KUBERNETES)
# tag::class[]
@Context
class SecretOperator:

    def __init__(self, shared_index_informer_factory: SharedIndexInformerFactory, controller_factory: ControllerFactory, namespace_resolver: NamespaceResolver):
        namespace = namespace_resolver.resolveNamespace()
        shared_index_informer_factory.sharedIndexInformerFor(V1Secret, namespace)
        controller_factory.createController(V1Secret, {namespace}, self.SecretReconciler())

    class SecretReconciler(ResourceReconciler[V1Secret]):
        def reconcile(self, request: Request, lister: OperatorResourceLister[V1Secret]) -> Result:
            secret_opt = lister.get(request)
            LOG.info("Reconciling secret: %s", request)
            # .. reconcile
            return Result(False)
# end::class[]

File: examples/example-kubernetes-client-openapi-operator-kotlin/src/main/kotlin/micronaut/operator/SecretOperator.kt
//tag::class[]
package micronaut.operator

import io.micronaut.context.annotation.Context
import io.micronaut.kubernetes.client.openapi.informer.SharedIndexInformerFactory
import io.micronaut.kubernetes.client.openapi.model.V1Secret
import io.micronaut.kubernetes.client.openapi.operator.OperatorResourceLister
import io.micronaut.kubernetes.client.openapi.operator.controller.ControllerFactory
import io.micronaut.kubernetes.client.openapi.operator.controller.reconciler.Request
import io.micronaut.kubernetes.client.openapi.operator.controller.reconciler.ResourceReconciler
import io.micronaut.kubernetes.client.openapi.operator.controller.reconciler.Result
import io.micronaut.kubernetes.client.openapi.resolver.NamespaceResolver
import org.slf4j.LoggerFactory
//end::class[]
import io.micronaut.context.annotation.Requires
import io.micronaut.context.env.Environment

@Requires(env = [Environment.KUBERNETES])
//tag::class[]

@Context
class SecretOperator(sharedIndexInformerFactory: SharedIndexInformerFactory, controllerFactory: ControllerFactory, namespaceResolver: NamespaceResolver) {

    companion object {
        private val LOG = LoggerFactory.getLogger(SecretOperator::class.java)
    }

    init {
        val namespace = namespaceResolver.resolveNamespace()
        sharedIndexInformerFactory.sharedIndexInformerFor(V1Secret::class.java, namespace)
        controllerFactory.createController(V1Secret::class.java, setOf(namespace), SecretReconciler())
    }

    private inner class SecretReconciler : ResourceReconciler<V1Secret> {
        override fun reconcile(request: Request, lister: OperatorResourceLister<V1Secret>): Result {
            val secretOpt = lister.get(request)
            LOG.info("Reconciling secret: {}", request)
            // .. reconcile
            return Result(false)
        }
    }
}
//end::class[]
